centreon/www/include/configuration/configObject/service_categories/DB-Func.php

```diff
@@ -156,64 +167,170 @@ function multipleServiceCategorieInDB($sc = [], $nbrDup = [])
+                            object_id: $maxId['maxid'],
+                            object_name: $sc_name,
+                            action_type: ActionLog::ACTION_TYPE_ADD,
+                            fields: $fields
+                        );
+                    } catch (CentreonDbException $ex) {
+                        CentreonLog::create()->error(
+                            logTypeId: CentreonLog::TYPE_BUSINESS_LOG,
+                            message: $ex->getMessage(),
+                            customContext: ['service_category_id' => $scId],
+                            exception: $ex,
+                        );
+
+                        throw $ex;
+                    }
+                }
             }
         }
     }
     CentreonACL::duplicateScAcl($scAcl);
     $centreon->user->access->updateACL();
 }
 
-function enableServiceCategorieInDB(?int $sc_id = null, $sc_arr = [])
+function enableServiceCategorieInDB(?int $serviceCategoryId = null, array $serviceCategories = [])
 {
-    if (!$sc_id && !count($sc_arr)) {
+
+    if (! $serviceCategoryId && empty($serviceCategories)) {
         return;
     }
-    global $pearDB;
-    if ($sc_id) {
-        $sc_arr = [$sc_id => "1"];
+
+    global $pearDB, $centreon;
+    if ($serviceCategoryId) {
+        $serviceCategories = [$serviceCategoryId => "1"];
     }
-    foreach (array_keys($sc_arr) as $key) {
-        $query = "UPDATE service_categories SET sc_activate = '1' WHERE sc_id = :sc_id";
-        $statement = $pearDB->prepare($query);
-        $statement->bindValue(':sc_id', $key, \PDO::PARAM_INT);
-        $statement->execute();
+
+    try {
+        $updateStatement = $pearDB->prepareQuery(
+            <<<SQL
+                UPDATE service_categories
+                SET sc_activate = '1'
+                WHERE sc_id = :serviceCategoryId
+            SQL
+        );
+        $selectStatement = $pearDB->prepareQuery(
+            <<<SQL
+                SELECT sc_name FROM `service_categories`
+                WHERE `sc_id` = :serviceCategoryId LIMIT 1
+            SQL
+        );
+        foreach (array_keys($serviceCategories) as $serviceCategoryId) {
+            $pearDB->executePreparedQuery($updateStatement, ['serviceCategoryId' => $serviceCategoryId]);
+            $pearDB->executePreparedQuery($selectStatement, ['serviceCategoryId' => $serviceCategoryId]);
+
+            $result = $pearDB->fetch($selectStatement);
+            $centreon->CentreonLogAction->insertLog(
+                object_type: ActionLog::OBJECT_TYPE_SERVICECATEGORIES,
+                object_id: $serviceCategoryId,
+                object_name: $result['sc_name'],
+                action_type: ActionLog::ACTION_TYPE_ENABLE
+            );
+        }
+    } catch(CentreonDbException $ex) {
+        CentreonLog::create()->error(
+            logTypeId: CentreonLog::TYPE_BUSINESS_LOG,
+            message: $ex->getMessage(),
```

Comment:
Not only the message of the exception, it's not a good practice, you need context too.

Reply:
good